src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -103,6 +104,10 @@ public function findSpecifiedType(
 							return null;
 						}
 
+						if ($needleType instanceof ConstantType && !$haystackType instanceof ConstantArrayType) {
```

Comment:
I tried different approaches on how to fix the added failling tests.

no matter which one I implement, some tests start failling. 
The PR as is fixed the newly added cases but regresses existing tests. 

I need to sleep over it

Reply:
Yes, `in_array` is a mad beast. It might need a complete rethinking of this approach.

This is also an opportunity to fix https://github.com/phpstan/phpstan/issues/5355 - it means that these conditions should not be in ImpossibleCheckTypeHelper, but in InArrayFunctionTypeSpecifyingExtension.